src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md

```diff
@@ -242,49 +263,55 @@ WARN[0013] Request Failed       error="Get http://test.k6.io: read tcp 172.31.72
 
 ### context deadline exceeded
 
-This happens when k6 was able to send a request, but the target system didn't respond in time. The default timeout in k6 is 60 seconds. If your system doesn't produce the response in this time frame, this error will appear.
+This happens when k6 could send a request, but the target system didn't respond in time. The default timeout in k6 is 60 seconds. If your system doesn't produce the response in this time frame, this error will appear.
```

Comment:
I prefer "was able to" here. It has more of an "ability" tone of something that happened in the past.

Reply:
How about "can"? I should've realized the real problem was the needless past tense.

Reply:
But the sentence is in past tense already, "can" wouldn't be a good fit either.

Why are you against "was able to"?

Reply:
The main clause is "This happens", which is present.

> Why are you against "was able to"?

Ha, yes, you've hit the heart of the issue.

"Is able to" longer than "can". That's the entire reason.

 I'm also generally against the past tense because it creates longer constructions and creates a grammatical distance that isn't necessary unless describing real pat tenses.

Reply:
But this is an extremely minor point. I'm fine with you choosing whatever you prefer, and merging when you think it's ready.

Reply:
We're really bikeshedding at this point :sweat_smile:, and I don't want to drag this on, but "this happens" is describing an action that happened in the past. I.e. you ran the test, got the error, and then "this happens when ...". You could write it either way, but describing a past action is more relatable to how the user would experience it.

Still, regardless of the tense, even the present "is able to" would be better than "can", since "can" implies that k6 has the _capability_ to do something, which, of course it does, it _can_ make requests; but "able to" implies the _ability_ in that moment, i.e. it's _unable_ to because of this or that reason.

But nevermind, it's fine as is. :smile:

Reply:
If you search google for "this {happens|occurs} when", you'll see most sentences are in the present tense. It's because the phrase introduces a general condition. In our sentences, no event _really_ occurred; we just describe the general conditions that create the error, as we understand them presently.

But, it's an interesting point about "can" vs. "is able to". I think in most cases, I'd only need "is unable to" but you've definitely got me questioning my "no-is-able-to" orthodoxy here.

I now realize the most direct way would be to remove all auxiliary verbs:

> This happens when k6 sends a request and the target system doesn't respond in time.

This particular case is not a big deal in the scheme of things, but I do appreciate any discussion that brings more attention to precision and clarity in language.

Reply:
Good points :+1: I agree that the simplest form is to remove all auxiliary verbs, as in your example there. Consistency is probably most important, and you have the big picture in mind. So I leave this up to you.

Thanks for the discussion :slightly_smiling_face: